src/main/java/com/epam/reportportal/cucumber/Utils.java

```diff
@@ -349,20 +353,20 @@ private static TestCaseIdEntry getTestCaseId(TestCaseId testCaseId, Method metho
 			}
 			return TestCaseIdUtils.getParameterizedTestCaseId(method, values.toArray());
 		} else {
-			return new TestCaseIdEntry(testCaseId.value(), testCaseId.value().hashCode());
+			return new TestCaseIdEntry(testCaseId.value());
 		}
 	}
 
 	private static TestCaseIdEntry getTestCaseId(String codeRef, List<cucumber.api.Argument> arguments) {
-		List<String> values = new ArrayList<String>(arguments.size());
-		for (cucumber.api.Argument argument : arguments) {
-			values.add(argument.getValue());
-		}
-		return new TestCaseIdEntry(StringUtils.join(codeRef, values.toArray()),
-				Arrays.deepHashCode(new Object[] { codeRef, values.toArray() })
-		);
+		return ofNullable(arguments).filter(args -> !args.isEmpty())
+				.map(args -> new TestCaseIdEntry(codeRef + TRANSFORM_PARAMETERS.apply(args)))
+				.orElseGet(() -> new TestCaseIdEntry(codeRef));
 	}
 
+	private static final Function<List<cucumber.api.Argument>, String> TRANSFORM_PARAMETERS = it -> "[" + it.stream()
+			.map(cucumber.api.Argument::getValue)
+			.collect(Collectors.joining(",")) + "]";
```

Comment:
Please use this collector (delimiter, prefix, suffix): 
`java.util.stream.Collectors#joining(java.lang.CharSequence, java.lang.CharSequence, java.lang.CharSequence)`

Reply:
fixed